packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -140,10 +141,15 @@ export class SummaryManager implements IDisposable {
         // enforce connectedState.clientId === clientElection.electedClientId. But once we're Running, we should
         // only transition to Stopping when the electedParentId changes. Stopping the summarizer without
         // changing the electedParent will just cause us to transition to Starting again.
-        if (this.connectedState.clientId !== this.clientElection.electedParentId ||
-            (this.state !== SummaryManagerState.Running &&
-                this.connectedState.clientId !== this.clientElection.electedClientId)) {
-            return { shouldSummarize: false, stopReason: "parentShouldNotSummarize" };
+
+        // Conditions for returning notElectedParent/notElectedClient and not running the last summary:
+        // a) New Parent has been elected and it is not the current client, or
+        // b) We are not already running the summarizer and we are not the current elected client id.
+        if (this.connectedState.clientId !== this.clientElection.electedParentId) {
```

Comment:
If I'm the summarizer client, would my this.connectedState.clientId be my parentId?

Reply:
I don't think it can be. electedClient == electedParent only until the client summarizer is spawn.